serde/src/java/org/apache/hadoop/hive/serde2/io/HiveCharWritable.java

```diff
@@ -83,8 +84,8 @@ public Text getStrippedValue() {
       return value;
     }
     // A lot of these methods could be done more efficiently by operating on the Text value
-    // directly, rather than converting to HiveChar.
-    return new Text(getHiveChar().getStrippedValue());
+    // directly, rather than converting to String
+    return new Text(StringUtils.stripEnd(value.toString(), " "));
```

Comment:
value.toString() in getHiveChar() would be the expensive part, due to decode. Ref: HIVE-24416.

Isn't "value.toString()" in the proposed patch have the same issue?

Reply:
yeah, you're absolutely right, this is just a best-effort to remove new HiveChar(), but that's not really the problem indeed...I'll think about eliminating decode

Reply:
also, I'm closing this and moving to HIVE-24416, which is the same